src/test/java/org/jabref/logic/importer/fetcher/ApsFetcherTest.java

```diff
@@ -0,0 +1,56 @@
+package org.jabref.logic.importer.fetcher;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.testutils.category.FetcherTest;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@FetcherTest
+class ApsFetcherTest {
+
+    private ApsFetcher finder;
+
+    @BeforeEach
+    void setUp() {
+        finder = new ApsFetcher();
+    }
+
+    private static Stream<Arguments> provideBibEntriesWithDois() throws IOException {
```

Comment:
To be honest, I don't see any need to have a parameterized test. As the comments indicate the tests actually check different cases and behaviours. It just happens that the test code is almost the same. Since the code reusage of the parameterized test is minimal, I would prefer if the tests could be split.